spec/classes/candlepin_spec.rb

```diff
@@ -107,7 +107,7 @@
         it do
           is_expected.to contain_yumrepo('candlepin')
             .with_descr('Candlepin: an open source entitlement management system.')
-            .with_baseurl("https://fedorapeople.org/groups/katello/releases/yum/nightly/candlepin/el7/\$basearch/")
+            .with_baseurl("https://fedorapeople.org/groups/katello/releases/yum/nightly/candlepin/"+facts[:osreleasemajor]+"/\$basearch/")
```

Comment:
@ekohl any suggestion here? I'm currently getting `no implicit conversion of nil into String`

Reply:
The fact doesn't exist. I believe structured facts only have symbols on the first level. You can also use string interpolation:
```suggestion
            .with_baseurl("https://fedorapeople.org/groups/katello/releases/yum/nightly/candlepin/#{facts[:os]['release']['major]}/\$basearch/")
```

Reply:
Thanks, trying this approach.